### PR TITLE
Translated Templatefields

### PR DESCRIPTION
**Pull request by nealmarques**
Fixes issue for when a templatefields are defined as translatable but no template is actually selected and the default record template has customfields.

I also added a fix for Bug: #110

### CODE REVIEW

**Comment by SahAssar** [comment]
Hey, Sorry it's taken me so long to look over this! Do you mind if I be a bit nitpicky on this? Otherwise I'll just fix it up in a separate branch, but I'll leave a few comments and lemme know.

File: src/EventListener/StorageListener.php [hunk 1]
```diff
@@ -99,7 +99,11 @@ public function preHydrate(HydrationEvent $event)
 				foreach ($localeData as $key => $value) {
 					if (isset($contentType['fields'][$key]['type']) && $contentType['fields'][$key]['type'] !== 'repeater') {
 						$subject[$key] = is_array($value) ? json_encode($value) : $value;
-					}
+					}  elseif ( $key == 'templatefields') {
```
Comment (SahAssar) [inline src/EventListener/StorageListener.php]: there are two spaces before `elseif` and one before $key, I try to sorta stick to PSR2

File: src/EventListener/StorageListener.php [hunk 2]
```diff
@@ -99,7 +99,11 @@ public function preHydrate(HydrationEvent $event)
...
+						foreach ($value as $i => $field) {
```
Comment (SahAssar) [inline src/EventListener/StorageListener.php]: `$i` is the field name here, right? a more descriptive naming would be nice, `$i` is usually just used for indexes on arrays

File: src/EventListener/StorageListener.php [hunk 3]
```diff
@@ -99,7 +99,11 @@ public function preHydrate(HydrationEvent $event)
...
+							$templatefields[$i]= is_array($field) ? json_encode($field) : $field;
```
Comment (SahAssar) [inline src/EventListener/StorageListener.php]: I see that there have snuck in tabs here and in a few other places (and I saw that there were a bunch of places they had snuck into the master repo too) could you replace them with 4 spaces and I'll clean up the ones in the core repo post-merge?

File: src/EventListener/StorageListener.php [hunk 4]
```diff
@@ -132,15 +136,21 @@ public function postHydrate(HydrationEvent $event)
...
-            if ($key === 'templatefields') {
-                $templateFields = $this->boltConfig->get('theme/templatefields/' . $subject['template'] . '/fields');
-                foreach ($templateFields as $key => $field) {
-                    if ($field['type'] === 'repeater') {
-                        $repeaterData = json_decode($value[$key], true);
-                        /** @var RepeatingFieldCollection[] $subject */
-                        $subject['templatefields'][$key]->clear();
-                        foreach ($repeaterData as $subValue) {
-                            $subject['templatefields'][$key]->addFromArray($subValue);
+            if ($key === 'templatefields' && !( $subject['template']==Null && !isset($contentType['record_template']) )) {
```
Comment (SahAssar) [inline src/EventListener/StorageListener.php]: Strict comparison and using the lowercase `null` constant would be nice, and the spacing is also a bit off (same thing in a few other places below). Something like `!($subject['template'] === null && !isset($contentType['record_template'])`

**Comment by nealmarques** [comment]
No worries,  please feel free to fix it up.  I just wanted to document what
I did to fix the issues I kept seeing.  I just finished up a big
translation job with all kinds of variations of content types with
specified templates, and non specified templates and default templates; so
I hope I ran into every kind of template combination translation scenario.
Take a look, I hope what I wrote makes sense.

Regards,

Neal

**Comment by SahAssar** [comment]
Merged via #132

**Pull request closed without merging**